### ISSUE TITLE
Replace/imrpove regex parsing for extracting co-authors of pr_collaborated events

### ISSUE THREAD

**Issue by rithviknishad** [ohcnetwork/leaderboard#492]
At times the pr_collaborated event yields incorrect co-author results

https://github.com/ohcnetwork/leaderboard/blob/ad4cb04bb06f1d51e952ab99263870dfcf08b30e/scraper/src/github-scraper/parseEvents.ts#L53-L55

**Comment by rithviknishad** [comment]
- https://github.com/coronasafe/leaderboard-data/blob/main/data/github/7.json
- https://github.com/coronasafe/leaderboard-data/blob/main/data/github/o.json

**Comment by vikram-8290** [comment]
hii @rithviknishad  i want to work on this issue i have understand that what is wrong with this but not to get that where should i make this chances , is there any documentation so the reference, is there any way you can help with this,
Thanks.

**Comment by rithviknishad** [comment]
https://docs.ohc.network/docs/leaderboard/gsoc-2024/refactor-scraper#3-parseeventsts

https://github.com/ohcnetwork/leaderboard/blob/ad4cb04bb06f1d51e952ab99263870dfcf08b30e/scraper/src/github-scraper/parseEvents.ts#L53-L55

**Comment by vikram-8290** [comment]
One thing I've identified is that the current implementation only captures a single match for the regex pattern /Co-authored-by: (.+) <(.+)>/, which can be problematic if there are multiple Co-authored-by entries in the commit message. Additionally, any slight variations in the format (like additional spaces or different capitalizations) could result in missed matches.

Proposed Fix:

Adjust the regular expression or use a more robust method to handle multiple Co-authored-by entries.
Implement a loop to capture all matches in the commit message rather than just the first occurrence.
Would you like me to implement this fix, or do you foresee any other potential issues to consider before moving forward?

**Comment by rithviknishad** [comment]
The core issue is, it's parsing authors incorrectly as mentioned in https://github.com/ohcnetwork/leaderboard/issues/492#issuecomment-2286338566

Finding all co-authors, nice find, it can simply be done by using `matchAll` instead of `match` right? Why create a loop and all?

**Comment by vikram-8290** [comment]
Thanks for pointing that out! You're absolutely right—using `matchAll` is a more straightforward and efficient way to handle this. It avoids the need for extra looping, as `matchAll` directly returns an iterator over all matches.

Should we consider making this change in issue #492?

**Comment by vikram-8290** [comment]
Yes, you’re correct that using `matchAll` is a great way to capture all co-authors in the commit message. However, we still need a loop to process each match individually. 

Here’s why the loop is necessary:

1. **Iterating Through Matches**: `matchAll` returns an iterator of all matches, and we need to loop through these matches to extract and process each co-author’s information.

2. **Processing Logic**: For each co-author, we need to check for conditions such as whether they are blacklisted or whether their names/emails are in our caches. The loop allows us to implement this logic for every individual match.

So, while `matchAll` helps us find all the matches, the loop is essential for handling the processing of each co-author found in the commit message.

If you're open to it, I can go ahead and implement this fix!

**Comment by vikram-8290** [comment]
I have created a pull request #519  . Please Check and let me Know if i have made some mistake.
Thanks

**Comment by Srayash** [comment]
hey @rithviknishad! waiting for review on #541.

**Comment by Utkarsh-Anandani** [comment]
@rithviknishad If this issue is still valid, I would like to work on it

**Comment by rithviknishad** [comment]
Comment how you are planning to solve this though

**Comment by rithviknishad** [comment]
leaderboard has been re-built, this is no longer valid. thanks for putting in efforts in trying to achieve to solve this!

github scraper has been moved out, however PR collaborated events are not yet tracked in the new scraper, feel free to check it out:

https://github.com/ohcnetwork/leaderboard-github-scraper
